File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;
class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        List<Integer> ans = new ArrayList<>();
        Queue<Integer> q = new LinkedList<>();
        
        for (int a : progresses) {
            q.offer(a);
        }
        
        int idx = 0;
        while(!q.isEmpty()) {
            int cnt = 1;
            
            int now = q.poll();
            int day = (100 - now);
            if (day % speeds[idx] != 0) day = day / speeds[idx] + 1;
            else day = day / speeds[idx];

            for (int i=idx+1;i<speeds.length;i++){
                int temp = q.peek();

                if (temp + speeds[i] * day >= 100) {
                    q.poll();
                    cnt++;
                    idx++;
                }
                else{
                    break;
                }
                
            }
            
            ans.add(cnt);
            idx++;
        }
        
        
        
        return ans.stream().mapToInt(x -> x).toArray();
    }
}

File: 프로그래머스/2/42586. 기능개발/기능개발.cpp
#include <bits/stdc++.h>

using namespace std;

queue<int> q;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;

    for (int i=0;i<progresses.size();i++) {
        int day = 0;
        int r = 100 - progresses[i];

        day += r / speeds[i];
        if (r % speeds[i]) day++;
        q.push(day);
    }

    int start = q.front();
    int temp = 0;

    while (q.size()) {
        if (q.front() <= start) {
            temp++;
            q.pop();
        } else {
            start = q.front();
            answer.push_back(temp);
            temp = 0;
        }
    }

    if (temp) answer.push_back(temp);





    return answer;
}